src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -186,4 +189,29 @@ private void setupTemplateAndPush(Repository repository, Resource[] resources, S
             repository.setFiles(null); // Clear cache to avoid multiple commits when ArTEMiS server is not restarted between attempts
         }
     }
+
+    // Generates the structure diff aka the test.json file for the current programming exercise
+    public void generateStructureDiffFile(URL solutionRepoURL, URL exerciseRepoURL, URL testRepoURL, String testsPath) throws IOException, InterruptedException {
+        Repository solutionRepository = gitService.getOrCheckoutRepository(solutionRepoURL);
+        Repository exerciseRepository = gitService.getOrCheckoutRepository(exerciseRepoURL);
+        Repository testRepository = gitService.getOrCheckoutRepository(testRepoURL);
+
+        gitService.pull(solutionRepository);
+        gitService.pull(exerciseRepository);
+        gitService.pull(testRepository);
+
+        String solutionRepositoryPath = solutionRepository.getLocalPath().toAbsolutePath().toString() + File.separator;
+        String exerciseRepositoryPath = exerciseRepository.getLocalPath().toAbsolutePath().toString() + File.separator;
+        String testRepositoryPath = testRepository.getLocalPath().toAbsolutePath().toString() + File.separator + testsPath;
+
+        StructureDiffGeneratorClient.run(solutionRepositoryPath, exerciseRepositoryPath, testRepositoryPath, "test.json");
+
```

Comment:
please properly create an object and invoke the method on this object. Do not use static methods if you can avoid it.